cabal-install/src/Distribution/Client/HttpUtils.hs

```diff
@@ -182,7 +183,11 @@ downloadURI transport verbosity uri path = do
     fragmentParser = do
         _ <- P.string "#sha256="
         str <- some P.hexDigit
+#if MIN_VERSION_base16_bytestring(1,0,0)
+        return (Base16.decodeLenient (BS8.pack str))
```

Comment:
So this is not quite equivalent behavior, because the previous behavior constructs valid bytestring "up to invalidity" in the source string, while `decodeLenient` passes through and does its best to construct a bytestring from what's left. Do we care about this in particular? It probably will not cause problems, but we should confirm.

Reply:
I do not see a good reason to use `decodeLenient` instead of at least `decode`. AFAIU ideally we should read an even number of hex digits into `str` (so that it's a valid base16 string), then parse with 
```
Right decoded <- Base16.decode (BS8.pack str)
return decoded
```
leveraging `instance MonadFail Parsec`.
(I do not have access rights to push necessary commits to this PR)

Reply:
That sounds more better to me if we want this to only accept morally correct inputs.

Reply:
@Bodigrim I added You to my fork, feel free to change it

Reply:
@emilypi  I don't think anybody cared about the validity of this bytestring. Code for older version of Base16-bytestring `return (fst (Base16.decode (BS8.pack str)))`  simply returns results without any care about error part of results tuple